CMakeLists.txt

```diff
@@ -28,15 +28,41 @@ target_include_directories(argparse INTERFACE
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>)
 
+
+# Pedantic compiler options
+# Update if necessary
+if(MSVC)
+  # Force to always compile with W4
+  if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
+    string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+  else()
+    set(ARGPARSE_PEDANTIC_COMPILE_FLAGS "/W4")
+  endif()
```

Comment:
Isn't just putting `/W4` will guarantee it will always replace other lower level warning level?